linuxdeploy-plugin-gtk.sh

```diff
@@ -177,11 +177,20 @@ HOOKSDIR="$APPDIR/apprun-hooks"
 HOOKFILE="$HOOKSDIR/linuxdeploy-plugin-gtk.sh"
 mkdir -p "$HOOKSDIR"
 cat > "$HOOKFILE" <<\EOF
-#! /bin/bash
+#! /usr/bin/env bash
 
 gsettings get org.gnome.desktop.interface gtk-theme 2> /dev/null | grep -qi "dark" && GTK_THEME_VARIANT="dark" || GTK_THEME_VARIANT="light"
 APPIMAGE_GTK_THEME="${APPIMAGE_GTK_THEME:-"Adwaita:$GTK_THEME_VARIANT"}" # Allow user to override theme (discouraged)
-CACHEDIR="$(mktemp -d /tmp/.AppRun.XXXXXXXX)"
+
+# in case we run from an AppImage, we use the $APPDIR environment variable as a template for the temporary directory that should be created
+# this allows users to attribute the tempdir to the running AppImage
+if [[ "$APPDIR" != "" ]]; then
```

Comment:
Use `if [ "$APPDIR" != "" ]; then` to be POSIX compliant here, because CI is failing: https://github.com/linuxdeploy/linuxdeploy-plugin-gtk/runs/4575332457?check_suite_focus=true

Reply:
True. Could probably also use `[ -z "$APPDIR" ]` or `[ -z $APPDIR ]` (never know which one is the right one...).

Reply:
Or `-n` (nonzero) like this:
```sh
if [ -n "$APPDIR" ]; then
    tempdir_template="$APPDIR".ld-p-gtk-tmp
else
    tempdir_template=/tmp/.ld-p-gtk-tmp-XXXXXX
fi
```
You need quotes, otherwise you will have this kind of message:
```
if [ -n $APPDIR ]; then
        ^-----^ SC2070: -n doesn't work with unquoted arguments. Quote or use [[ ]].
        ^-----^ SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
Yep, saw it while trying those locally. I used your suggested syntax, though.

Reply:
Please commit the change, then I will approve this PR. :wink:

Reply:
fwiw, the safest way to use variables in sh is `"${VARIABLE}"`. I am trying to use this format where I can.

Reply:
I actually have committed the fix a couple days ago. https://github.com/linuxdeploy/linuxdeploy-plugin-gtk/pull/26/commits/4a1fd0a25bd49a777aa1ab25b01b616ccd30b070

Reply:
@X0rg please re-review.

Reply:
LGTM.